### ISSUE TITLE
rgdal build with gdal from core but not with gdal 2 from osgeo4mac

### ISSUE THREAD

**Issue by luisspuerto** [OSGeo/homebrew-osgeo4mac#370]
Hi! 

I tried to update update rgdal package on R and came across this error: 

```sh
trying URL 'cloud.r-project.org/src/contrib/rgdal_1.2-20.tar.gz'
Content type 'application/x-gzip' length 1650449 bytes (1.6 MB)
==================================================
downloaded 1.6 MB

* installing *source* package ‘rgdal’ ...
** package ‘rgdal’ successfully unpacked and MD5 sums checked
configure: CC: /usr/local/opt/llvm/bin/clang
configure: CXX: /usr/local/opt/llvm/bin/clang++
configure: rgdal: 1.2-20
checking for /usr/bin/svnversion... yes
configure: svn revision: 725
checking for gdal-config... /usr/local/bin/gdal-config
checking gdal-config usability... yes
configure: GDAL: 2.2.4
checking GDAL version >= 1.6.3... yes
checking gdal: linking with --libs only... yes
checking GDAL: /usr/local/Cellar/gdal2/2.2.4/share/gdal/pcs.csv readable... yes
configure: -g -O3 -Wall -pedantic -std=gnu99 -mtune=native -pipe  -I/usr/local/Cellar/gdal2/2.2.4/include
checking proj_api.h presence and usability... yes
dyld: Library not loaded: /usr/local/opt/cfitsio/lib/libcfitsio.5.dylib
  Referenced from: /usr/local/opt/gdal2/lib/libgdal.20.dylib
  Reason: image not found
checking projects.h presence and usability... yes
checking PROJ.4 version: ... ./configure: line 2055: test: -eq: unary operator expected
./configure: line 2060: test: -ge: unary operator expected
< 4.8.0
./configure: line 2087: test: -ge: unary operator expected
proj_conf_test2.c:3:7: error: conflicting types for 'pj_open_lib'
FILE *pj_open_lib(const char *, const char *);
      ^
/usr/local/include/proj_api.h:204:8: note: previous declaration is here
PAFile pj_open_lib(projCtx, const char *, const char *);
       ^
proj_conf_test2.c:7:34: error: too few arguments to function call, expected 3, have 2
    fp = pj_open_lib("epsg", "rb");
         ~~~~~~~~~~~             ^
/usr/local/include/proj_api.h:204:1: note: 'pj_open_lib' declared here
PAFile pj_open_lib(projCtx, const char *, const char *);
^
2 errors generated.
./configure: line 2134: ./proj_conf_test2: No such file or directory
checking PROJ.4: epsg found and readable... yes
./configure: line 2162: test: -ge: unary operator expected
proj_conf_test3.c:3:7: error: conflicting types for 'pj_open_lib'
FILE *pj_open_lib(const char *, const char *);
      ^
/usr/local/include/proj_api.h:204:8: note: previous declaration is here
PAFile pj_open_lib(projCtx, const char *, const char *);
       ^
proj_conf_test3.c:7:35: error: too few arguments to function call, expected 3, have 2
    fp = pj_open_lib("conus", "rb");
         ~~~~~~~~~~~              ^
/usr/local/include/proj_api.h:204:1: note: 'pj_open_lib' declared here
PAFile pj_open_lib(projCtx, const char *, const char *);
^
2 errors generated.
./configure: line 2209: ./proj_conf_test3: No such file or directory
checking PROJ.4: conus found and readable... yes
configure: Package CPP flags:  -I/usr/local/Cellar/gdal2/2.2.4/include
configure: Package LIBS:  -L/usr/local/Cellar/gdal2/2.2.4/lib -lgdal -lproj
configure: creating ./config.status
config.status: creating src/Makevars
** libs
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.5.0_1/lib/R/include" -DNDEBUG -I/usr/local/Cellar/gdal2/2.2.4/include -I"/Users/lpuerto/Library/R/3.x/library/sp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/llvm/include   -fPIC  -I/Library/Java/JavaVirtualMachines/jdk-10.0.1.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-10.0.1.jdk/Contents/Home/include/darwin -c OGR_write.cpp -o OGR_write.o
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.5.0_1/lib/R/include" -DNDEBUG -I/usr/local/Cellar/gdal2/2.2.4/include -I"/Users/lpuerto/Library/R/3.x/library/sp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/llvm/include   -fPIC  -I/Library/Java/JavaVirtualMachines/jdk-10.0.1.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-10.0.1.jdk/Contents/Home/include/darwin -c gdal-bindings.cpp -o gdal-bindings.o
/usr/local/opt/llvm/bin/clang  -I"/usr/local/Cellar/r/3.5.0_1/lib/R/include" -DNDEBUG -I/usr/local/Cellar/gdal2/2.2.4/include -I"/Users/lpuerto/Library/R/3.x/library/sp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/llvm/include   -fPIC  -g -O3 -Wall -pedantic -std=gnu99 -mtune=native -pipe -c init.c -o init.o
/usr/local/opt/llvm/bin/clang  -I"/usr/local/Cellar/r/3.5.0_1/lib/R/include" -DNDEBUG -I/usr/local/Cellar/gdal2/2.2.4/include -I"/Users/lpuerto/Library/R/3.x/library/sp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/llvm/include   -fPIC  -g -O3 -Wall -pedantic -std=gnu99 -mtune=native -pipe -c inverser.c -o inverser.o
/usr/local/opt/llvm/bin/clang  -I"/usr/local/Cellar/r/3.5.0_1/lib/R/include" -DNDEBUG -I/usr/local/Cellar/gdal2/2.2.4/include -I"/Users/lpuerto/Library/R/3.x/library/sp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/llvm/include   -fPIC  -g -O3 -Wall -pedantic -std=gnu99 -mtune=native -pipe -c local_stubs.c -o local_stubs.o
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.5.0_1/lib/R/include" -DNDEBUG -I/usr/local/Cellar/gdal2/2.2.4/include -I"/Users/lpuerto/Library/R/3.x/library/sp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/llvm/include   -fPIC  -I/Library/Java/JavaVirtualMachines/jdk-10.0.1.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-10.0.1.jdk/Contents/Home/include/darwin -c ogr_geom.cpp -o ogr_geom.o
/usr/local/opt/llvm/bin/clang  -I"/usr/local/Cellar/r/3.5.0_1/lib/R/include" -DNDEBUG -I/usr/local/Cellar/gdal2/2.2.4/include -I"/Users/lpuerto/Library/R/3.x/library/sp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/llvm/include   -fPIC  -g -O3 -Wall -pedantic -std=gnu99 -mtune=native -pipe -c ogr_polygons.c -o ogr_polygons.o
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.5.0_1/lib/R/include" -DNDEBUG -I/usr/local/Cellar/gdal2/2.2.4/include -I"/Users/lpuerto/Library/R/3.x/library/sp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/llvm/include   -fPIC  -I/Library/Java/JavaVirtualMachines/jdk-10.0.1.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-10.0.1.jdk/Contents/Home/include/darwin -c ogr_proj.cpp -o ogr_proj.o
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.5.0_1/lib/R/include" -DNDEBUG -I/usr/local/Cellar/gdal2/2.2.4/include -I"/Users/lpuerto/Library/R/3.x/library/sp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/llvm/include   -fPIC  -I/Library/Java/JavaVirtualMachines/jdk-10.0.1.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-10.0.1.jdk/Contents/Home/include/darwin -c ogrdrivers.cpp -o ogrdrivers.o
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.5.0_1/lib/R/include" -DNDEBUG -I/usr/local/Cellar/gdal2/2.2.4/include -I"/Users/lpuerto/Library/R/3.x/library/sp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/llvm/include   -fPIC  -I/Library/Java/JavaVirtualMachines/jdk-10.0.1.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-10.0.1.jdk/Contents/Home/include/darwin -c ogrsource.cpp -o ogrsource.o
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.5.0_1/lib/R/include" -DNDEBUG -I/usr/local/Cellar/gdal2/2.2.4/include -I"/Users/lpuerto/Library/R/3.x/library/sp/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/llvm/include   -fPIC  -I/Library/Java/JavaVirtualMachines/jdk-10.0.1.jdk/Contents/Home/include -I/Library/Java/JavaVirtualMachines/jdk-10.0.1.jdk/Contents/Home/include/darwin -c projectit.cpp -o projectit.o
clang++ -std=gnu++11 -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/usr/local/opt/gettext/lib -L/usr/local/opt/llvm/lib -Wl,-rpath,/usr/local/opt/llvm/lib -L/usr/local/Cellar/r/3.5.0_1/lib/R/lib -L/usr/local/opt/gettext/lib -L/usr/local/opt/llvm/lib -Wl,-rpath,/usr/local/opt/llvm/lib -o rgdal.so OGR_write.o gdal-bindings.o init.o inverser.o local_stubs.o ogr_geom.o ogr_polygons.o ogr_proj.o ogrdrivers.o ogrsource.o projectit.o -L/usr/local/Cellar/gdal2/2.2.4/lib -lgdal -lproj -L/usr/local/Cellar/r/3.5.0_1/lib/R/lib -lR -lintl -Wl,-framework -Wl,CoreFoundation
installing to /Users/lpuerto/Library/R/3.x/library/rgdal/libs
** R
** data
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
Error: package or namespace load failed for ‘rgdal’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Users/lpuerto/Library/R/3.x/library/rgdal/libs/rgdal.so':
  dlopen(/Users/lpuerto/Library/R/3.x/library/rgdal/libs/rgdal.so, 6): Library not loaded: /usr/local/opt/cfitsio/lib/libcfitsio.5.dylib
  Referenced from: /usr/local/opt/gdal2/lib/libgdal.20.dylib
  Reason: image not found
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/Users/lpuerto/Library/R/3.x/library/rgdal’
* restoring previous ‘/Users/lpuerto/Library/R/3.x/library/rgdal’

The downloaded source packages are in
	‘/private/var/folders/wf/41gjf2mx7m7fmvfd8dr22_5h0000gn/T/RtmpjWtI0L/downloaded_packages’
Warning message:
In install.packages(update[instlib == l, "Package"], l, contriburl = contriburl,  :
  installation of package ‘rgdal’ had non-zero exit status
```

I had linked gdal2 as my main gdal. So I decided to unlink it and return to the main gdal from core. 

Now I'm able to build the package without a problem.

**Comment by nickrobison** [comment]
@luisspuerto Sorry to hear that you're having issues.

I'm not sure how well having both gdal instances installed (event unlinked) will work for you going forward. Is there a reason that you need both?

The error message that you're seeing can sometimes happen if a package gets upgraded in the background and the new library changes a bit.

Can you post the output of:

`ls -la /usr/local/opt/gdal2/lib/`?

**Comment by luisspuerto** [comment]
PS/ i don't know if I need them both, but as far as I know yes. 

homebrew formulae build agains `gdal`, while `qgis2` build agains `gdal2`. 

I know we've discussed about have only `gdal` from `core` in the near future, but I think that it's not going to workout well since [they seem to head](https://discourse.brew.sh/t/r-bottle-options-graphics-capabilities/1785/10) to an option-less homebrew-core. I just found out that R from Homebrew has less capabilities than the binary from the original developer.
